doc/contributing.md

```diff
@@ -0,0 +1,10 @@
+# Contributing to ANTLR 4
```

Comment:
:question: Do we want to make this a **CONTRIBUTING.md** file in the root of the repository? GitHub recognizes that as a special file that's presented to the user in some parts of the UI.

(I really mean this as a question and not action item since it's subjective/preference either way)